### PR TITLE
Check for jobs going held with condor-g

### PR DESCRIPTION
**Pull request by efajardo**
@matyasselmeci this is a bug fix/ feaature since the grid probes were not checking for the jobs going held. Which is the most common problem in the grid. I added support for this. Can you take a look and let me know what you think?

### CODE REVIEW

File: rsv-core/lib/python/rsv/Results.py [hunk 1]
```diff
@@ -380,6 +380,15 @@ def condor_g_remote_gatekeeper_down(self, metric, log):
         data  += "Log:\n%s\n" % log
 
         self.brief_result(metric, status, data, stderr="")
+    
+    def job_was_held(self, metric, log):
+        """ Condor-G submissioned failed because the job went to held """
```
Comment (matyasselmeci) [inline rsv-core/lib/python/rsv/Results.py]: "went to held" should be "went on hold".

File: rsv-core/lib/python/rsv/Results.py [hunk 2]
```diff
@@ -380,6 +380,15 @@ def condor_g_remote_gatekeeper_down(self, metric, log):
...
+        status = "CRITICAL"
+        data   = "Condor-G submission failed the job went to held.\n"
+        data  += "Check the log and the held reason.\n\n"
```
Comment (matyasselmeci) [inline rsv-core/lib/python/rsv/Results.py]: "held reason" should be "hold reason".

File: rsv-core/lib/python/rsv/Results.py [hunk 3]
```diff
@@ -380,6 +380,15 @@ def condor_g_remote_gatekeeper_down(self, metric, log):
         data  += "Log:\n%s\n" % log
 
         self.brief_result(metric, status, data, stderr="")
+    
+    def job_was_held(self, metric, log):
+        """ Condor-G submissioned failed because the job went to held """
+        status = "CRITICAL"
+        data   = "Condor-G submission failed the job went to held.\n"
```
Comment (matyasselmeci) [inline rsv-core/lib/python/rsv/Results.py]: "went to held" should be "went on hold".

**Comment by matyasselmeci** [comment]
Other than a few grammar problems that I highlighed in line notes, it looks good to me.

**Comment by efajardo** [comment]
@matyasselmeci I fixed the grammar errors. Can you take a look again please?

**Pull request merged**